src/index.ts

```diff
@@ -138,95 +148,390 @@ export = (app: Probot) => {
+          const issueItemID = await GraphQL.getIssueItemID(projectID, issueNumber);
+          const issueItemStatus = await GraphQL.getIssueItemStatus(projectID, issueNumber);
+
+          if (issueItemStatus != "🏗 In progress") {
+            throw new Error('Issue is not in "🏗 In progress" status. Can\'t change status to "👀 In review".');
+          } else {
+            GraphQL.changeItemStatus(projectID, issueItemID, "👀 In review");
+          }
+
+          // Change status of previous linked issue to "🏗 In progress"
+          const previousIssueItemID = await GraphQL.getIssueItemID(projectID, previousIssueNumber);
+          const previousIssueItemStatus = await GraphQL.getIssueItemStatus(projectID, previousIssueNumber);
+
+          if (previousIssueItemStatus != "👀 In review") {
+            throw new Error("Previous issue is not in 👀 In review status. Can't change status to 🏗 In progress.");
+          } else {
+            GraphQL.changeItemStatus(projectID, previousIssueItemID, "🏗 In progress");
+          }
+        }
+      } else {
+        throw new Error("Previous comment is undefined. Issue remains in 👀 In review status.");
+      }
+    } else {
+      throw new Error("PR is not linked to an issue.");
+    }
+  });
+
+  // For “👀 In review” issues, change status to “🏗 In progress” when PR is closed without merging
+  // For “👀 In review” issues, change status to “Done” when PR is merged
+  app.on("pull_request.closed", async (context) => {
+    const comment = context.payload.pull_request.body?.trim();
+    const pullRequestNumber = context.payload.pull_request.number;
+    const pullRequestRepo = context.payload.pull_request.head.repo.name;
+
+    if (linksToIssue(comment)) {
+      const projectID = await GraphQL.getProjectID(org, projectNumber);
+      const issueNumber = await GraphQL.getLinkedIssueNumber(projectID, pullRequestNumber, pullRequestRepo);
+      const wasMerged = context.payload.pull_request.merged;
+
+      const issueItemID = await GraphQL.getIssueItemID(projectID, issueNumber);
+      const issueItemStatus = await GraphQL.getIssueItemStatus(projectID, issueNumber);
+
+      if (issueItemStatus == "👀 In review" && wasMerged) {
+        // Change status to "✅ Done" if PR was merged
+        GraphQL.changeItemStatus(projectID, issueItemID, "✅ Done");
+      } else if (issueItemStatus == "👀 In review" && !wasMerged) {
+        // Change status to "🏗 In progress" if PR was closed without merging
+        GraphQL.changeItemStatus(projectID, issueItemID, "🏗 In progress");
+      } else if (issueItemStatus == "✅ Done" && wasMerged) {
+        console.log('Default GitHub Workflow already changed status to "✅ Done". Nothing to do.');
+      } else {
+        throw new Error("Issue is in incorrect status. Can't change status.");
+      }
+    } else {
+      throw new Error("PR is not linked to an issue.");
+    }
+  });
 
   // Update contributors table in README.md based on contributors.json in main
```

Comment:
let's keep it together with contributors-related commands parsing. or maybe let's split this two bot functions into two different files